ambari-server/src/main/resources/common-services/SPARK2/2.0.0/package/scripts/alerts/alert_spark2_thrift_port.py

```diff
@@ -131,9 +148,16 @@ def execute(configurations={}, parameters={}, host_name=None):
             host_name = socket.getfqdn()
 
         if security_enabled:
-            beeline_url = ["'jdbc:hive2://{host_name}:{port}/default;principal={hive_principal}'","transportMode={transport_mode}"]
+            if spark_ssl_enabled:
+                beeline_url = ['"jdbc:hive2://{host_name}:{port}/default;principal={hive_principal};transportMode={transport_mode;ssl=true;sslTrustStore={spark_truststore_path};trustStorePassword={spark_truststore_pass!p};httpPath=cliservice"']
+            else:
+                beeline_url = ["jdbc:hive2://{host_name}:{port}/default;principal={hive_principal}","transportMode={transport_mode}"]
         else:
-            beeline_url = ["'jdbc:hive2://{host_name}:{port}/default'","transportMode={transport_mode}"]
+            if spark_ssl_enabled:
```

Comment:
These nested if-else blocks could be re-written to be less nested (e.g. by using stringbuilder). That would also reduce the chances of bugs like mentioned in the previous comment for line 152.

Reply:
i would like work on that re-write part as part of separate BUG as there is one more similar BUG.

Reply:
Sounds good.